### ISSUE TITLE
Help accessing a public site that has a locked gallery

### ISSUE THREAD

**Issue by ricknegrin** [AlexGhiondea/SmugMug.NET#25]
I am trying to access a SmugMug site that is open to the public (so there is no need to authenticate) but has a gallery that is locked with a password.  Is there a way to access that using SmugMug.NET? It looks like the site object is tied to a user object.  Any help would be appreciated. Thanks.

**Comment by AlexGhiondea** [comment]
That is a great question. I thought you could authenticate without a password, but I guess that was deprecated by the move to oAuth. 

At this time, the only think I can think of is trying to login with any account and then try to access the password protected album.

**Comment by ricknegrin** [comment]
Alex,

Thanks for responding.  I am not sure what you mean.  The object model only allows me to get albums based on my account (i.e. the user object returned from the Login function on the Site object).  How can I specify a specific site that is not the one bound to my user?  I have been going through the code and I don't see a way to do that.  (BTW, I found a parameter for the site password in the GetAlbums override function, so that seems promsing. I just have to figure out how to reference the site/Album I want.

**Comment by AlexGhiondea** [comment]
You are correct -- there is no good way to get to the album if the album is not yours and password protected :(.

The only thing I can think of is trying to make the request yourself.

You can either try and re-use the CommunicationHelper class and write a method that will access any random album, or try to write this at the HttpClient level with a GET request. You can inspect the SmugMug APIs here: http://api.smugmug.com/api/v2.

**Comment by ricknegrin** [comment]
I see. Thank you for the help.
